crates/ide_diagnostics/src/handlers/remove_unnecessary_else.rs

```diff
@@ -0,0 +1,207 @@
+use ide_db::source_change::SourceChange;
+use syntax::{
+    ast::{self, ElseBranch},
+    AstNode, SyntaxNode, TextRange,
+};
+use text_edit::TextEdit;
+
+use crate::{fix, Diagnostic, FileId, Severity};
+
+// Diagnostic: remove-unnecessary-else
+//
+// This diagnostic is triggered when we have an `else` block following an `if` block which ends in a `return` statement.
+// (or a `continue`/`break` if inside a loop).
+pub(crate) fn unnecessary_else(
+    acc: &mut Vec<Diagnostic>,
+    file_id: FileId,
+    node: &SyntaxNode,
+) -> Option<()> {
+    let if_expr = ast::IfExpr::cast(node.clone())?;
+    if let Some(else_branch) = if_expr.else_branch() {
+        match else_branch {
+            ElseBranch::Block(else_block) => {
```

Comment:
```suggestion
            Some(ElseBranch::Block(else_block)) => {
```

(to collapse the `if let` and `match`)